roles/StackStorm.nginx/tasks/nginx_yum.yml

```diff
@@ -40,10 +44,64 @@
     state: absent
   tags: nginx
 
+- name: Backup nginx.conf
+  become: yes
+  copy:
+     remote_src: yes
+     src: /etc/nginx/nginx.conf
+     dest: /etc/nginx/nginx.conf.bak
+  tags: nginx
+  when: (ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '8')
+
+- name: Comment out server block in nginx.conf temporary file
+  become: yes
+  shell:
+     cmd: awk '/^    server {/{f=1}f{$0 = "#" $0}{print}' /etc/nginx/nginx.conf.bak > /tmp/nginx.conf
+  args:
+      warn: False
+  # Use awk to match one line install
+  tags: [nginx, skip_ansible_lint]
+  when: (ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '8')
+
+- name: Restore nginx.conf from temporary file
+  become: yes
+  copy:
+     remote_src: yes
+     dest: /etc/nginx/nginx.conf
+     src: /tmp/nginx.conf
+  tags: nginx
+  when: (ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '8')
+
+- name: Delete temporary file
+  become: yes
+  file:
+     path: /tmp/nginx.conf
+     state: absent
+  tags: nginx
+  when: (ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '8')
+
+- name: Cleanup double comments
+  become: yes
+  replace:
+     regexp: "##"
+     replace: "#"
+     path: /etc/nginx/nginx.conf
+  tags: nginx
+  when: (ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '8')
+
+- name: Cleanup comment closing out server block
```

Comment:
These set of tasks coming from bash needs a proper re-write by using Ansible powers better.
One-line bash/awk/tmp hacks are coming from the fact it was the best way to do it for curl-bash installer.

The objective here is simple: disable default nginx record which is conflicting with the st2 nginx config.

Reply:
> These set of tasks coming from bash needs a proper re-write by using Ansible power better.
> One-line bash/awk/tmp hacks are coming from the fact it was the best way to do it for curl-bash installer.
> 
> The objective here is simple: disable default nginx record which is conflicting with the st2 nginx config.

Have that fixed replaced with 2 replace calls instead (couldn't get a single one to do it). Need to fix some idempotent issues with the yum module enable/disable calls which aren't supported in ansible yum's module. Will put up changes once I've got those fixed and re-run. Probably early next week now though.

Reply:
@amanda11 - this is kick ass! thanks for your help on this!